### ISSUE TITLE
Stuck in Invalid Handle

### ISSUE THREAD

**Issue by serghidalg** [bluesky-social/pds#133]
Hey there!
I have tried setting up the PDS configuration and the only divergence I have is that I host Caddy in a different machine so I just do the reverse proxy from there to port 3000 in the PDS machine.

Registration works fine and I can see my profile at https://bsky.app/profile/computers.pyam.org
However, there is always a Invalid Handle note, meaning that something is not correctly set up.

What I have tried so far:

- CURL to update handle and change it from the website -> Handle changes but is still invalid.
- Checking for A and TXT in my DNS -> https://pyam.org seems to be working okay and contains all the needed info.
- Use Bluesky Debug Page [here](https://bsky-debug.app/handle?handle=pyam.org) -> Everything seems to be working fine

This last check is what bugs me the most since I can do everything fine but I still get the error and I cannot be found by other accounts.

Could you please guide through the debugging of the setup?

## TLDR: Stuck in Invalid Handle with PDS setup even though registration process works fine.

**Comment by fullStackRacc** [comment]
Having the same issue and I don't know if it's the site being weird because of all the new people or not, curious if anyone else knows or has troubleshooting steps

**Comment by fullStackRacc** [comment]
Okay so I just figured out why mine was doing it so I have to check, did you add the wildcard entry in your DNS? like *.pyam.org

**Comment by serghidalg** [comment]
> Okay so I just figured out why mine was doing it so I have to check, did you add the wildcard entry in your DNS? like *.pyam.org

The wirdcard was added. As I said in my second troubleshooting step, that is correctly set up. 
I just double checked in DNS checker and it shows that any domain in *.pyam.org works fine.

Not sure what is making it not work

**Comment by fullStackRacc** [comment]
yeah I just wanted to double check, i had forgotten to haha 

hopefully in that case it's just the site being a bit glitchy with the new people, but if not hopefully someone with more information responds and can get you situated, it's super annoying

**Comment by slevengood** [comment]
I was having this same issue earlier, but after just waiting a bit and trying again it ended up working. I think things may just be a little slow/weird due to the influx of new users.

**Comment by serghidalg** [comment]
I just noticed that if I explicitly set up the DNS TXT for a certain user, the Invalid Handle dissappears.
However I thought that if they were hosted in your PDS, this process was implicit and if a new handle was used, you didn't need to set it up manually in your DNS. Was I mistaken?
i.e: computers.pyam.org will not work untir I add hhe did to the DNS TXT record. If I later decide to change the handle for that user, I will need to set up a new TXT record and same goes for any newly created accounts.

Do you update by hand the DNS TXT record for each new handle a user creates?

**Comment by mcangeli** [comment]
Yes, you have to add the _atproto txt entry in DNS for each user.

**Comment by serghidalg** [comment]
So how do huge servers handle this?
Isn't there a way to automate thin process?
What happens if a user changes their handle down the line?

**Comment by bbeallo12** [comment]
I'm having this issue too.
Both the DNS and http check are reporting as working.
So I'm not sure what the issue could be.

**Comment by t0mtaylor** [comment]
I found that posting from an account once i fixed the verifications, it removed the `invalid handle` error on the profile

**Comment by fullStackRacc** [comment]
im gonna be completely honest i thought i closed this one too, it ended up being the dns acting bananas iirc

**Comment by mpermar** [comment]
> Yes, you have to add the _atproto txt entry in DNS for each user.

Although this is true, note that the point of including Caddy with the PDS is that it is meant to handle certificate creation, verification and renewal automatically for every user. 

I was hit by this `Invalid handle` error as well and after a little bit of digging I found that the culprit was Caddy which somehow had got corrupted in a way like what is [mentioned in this thread](https://github.com/caddyserver/caddy/issues/6481), and you could watch errors like this in the pod logs when trying to set then handle from the bsky UI.

```
caddy       | {"level":"error","ts":1736598547.3427942,"logger":"tls.on_demand","msg":"loading newly-obtained certificate from storage","remote_ip":"146.59.1.35","remote_port":"59318","server_name":"martin.arume.cloud","error":"no matching certificate to load for martin.arume.cloud: decoding certificate metadata: invalid character '}' after top-level value"}
```

Wiping out the certificates folder in caddy and restarting the pods made Caddy back happy and fixed the issue for me.

**Comment by serghidalg** [comment]
That solves the issue then. Thanks!
